spark_pipeline_framework/transformers/fhir_sender/v1/fhir_sender.py

```diff
@@ -364,6 +383,32 @@ def _transform(self, df: DataFrame) -> DataFrame:
                 desired_partitions: int = num_partitions or math.ceil(
                     row_count / batch_size
                 )
+                if sort and sort.get("partition_by_column_name"):
+                    json_df.withColumn(
+                        sort["partition_by_column_name"],
+                        get_json_object(
+                            col("value"), f"$.{sort['partition_by_column_name']}"
+                        ),
+                    ).repartition(sort["partition_by_column_name"])
+                    if sort.get("column_for_sorting"):
+                        json_df.sortWithinPartitions(sort["column_for_sorting"])
+                    json_df = json_df.drop(sort["partition_by_column_name"])
+                    desired_partitions = json_df.rdd.getNumPartitions()
+
+                if sort and sort.get("drop_column"):
+                    # removing field from all the json values
+                    remove_field_from_json_udf = udf(
+                        lambda json_data: remove_field_from_json(
+                            json_data, sort["column_for_sorting"]
+                        ),
+                        StringType(),
+                    )
+                    json_df = json_df.withColumn(
+                        "value", remove_field_from_json_udf(col("value"))
+                    )
+                    # dropping column which was added to sort the data within partitions
+                    json_df = json_df.drop(sort["column_for_sorting"])
```

Comment:
may not be required depending on other comments

Reply:
This will still be required because we're creating new column for sorting within partitions.